benchmark/src/stats.rs

```diff
@@ -0,0 +1,83 @@
+use serde::Serialize;
+
+/// Statistics for a a single test on a program
+#[derive(Debug, Clone, Serialize)]
+pub struct TestResult {
+    pub filename: String,
+    pub passed: bool,
+}
+
+/// Statistics for running many tests on a single program
+#[derive(Debug, Serialize)]
+pub struct ProgramEvalStats {
+    pub program_name: String,
+    pub translation_success: bool,
+    pub rust_build_success: bool,
+    pub total_tests: usize,
+    pub passed_tests: usize,
+    pub error_message: Option<String>,
+    // Store individual test results with filenames and pass/fail status
+    pub test_results: Vec<TestResult>,
+}
+
+impl ProgramEvalStats {
+    /// Calculate success rate as a percentage
+    pub fn success_rate(&self) -> f64 {
+        if self.total_tests == 0 {
+            0.0
+        } else {
+            (self.passed_tests as f64 / self.total_tests as f64) * 100.0
+        }
```

Comment:
Nitpick: IMO these would be cleaner as a `match` statement (or an `if` that returns early if `total_tests` is 0 followed by the floating-point math).